test/test_deconvolutional_layer.h

```diff
@@ -42,158 +41,125 @@ TEST(deconvolutional, setup_nnp) {
                           backend_t::nnpack);
 
   EXPECT_EQ(l.parallelize(), true);          // if layer can be parallelized
-  EXPECT_EQ(l.in_channels(), 3u);            // num of input tensors
+  EXPECT_EQ(l.in_channels(), 1u);            // num of input tensors
   EXPECT_EQ(l.out_channels(), 1u);           // num of output tensors
   EXPECT_EQ(l.in_data_size(), 4u);           // size of input tensors
   EXPECT_EQ(l.out_data_size(), 32u);         // size of output tensors
   EXPECT_EQ(l.in_data_shape().size(), 1u);   // number of inputs shapes
   EXPECT_EQ(l.out_data_shape().size(), 1u);  // num of output shapes
-  EXPECT_EQ(l.weights().size(), 2u);         // the wieghts vector size
-  EXPECT_EQ(l.weights_grads().size(), 2u);   // the wieghts vector size
-  EXPECT_EQ(l.inputs().size(), 3u);          // num of input edges
+  EXPECT_EQ(l.inputs().size(), 1u);          // num of input edges
   EXPECT_EQ(l.outputs().size(), 1u);         // num of outpus edges
-  EXPECT_EQ(l.in_types().size(), 3u);        // num of input data types
+  EXPECT_EQ(l.in_types().size(), 1u);        // num of input data types
   EXPECT_EQ(l.out_types().size(), 1u);       // num of output data types
   EXPECT_EQ(l.fan_in_size(), 9u);            // num of incoming connections
   EXPECT_EQ(l.fan_out_size(), 18u);          // num of outgoing connections
+  EXPECT_EQ(l.parameters().size(), 2u);      // num of trainable parameters
+  EXPECT_EQ(l.weights_at(0).size(), 18u);    // size of weight parameter
+  EXPECT_EQ(l.biases_at(0).size(), 2u);      // size of bias parameter
   EXPECT_STREQ(l.layer_type().c_str(), "deconv");  // string with layer type
 }
 #endif
 
 TEST(deconvolutional, fprop) {
-  network<sequential> nn;
-
   deconvolutional_layer l(2, 2, 3, 1, 2);
+  l.weight_init_f(parameter_init::constant(0.0));
```

Comment:
It is conflicting with existing method of same name and since quantization layers still don't have parameters, I need to keep both methods. I have added a todo to remove old method and rename this one once all layers have parameters integrated or once quantization disabling PR is merged and this one is rebased on top of it.